custom_components/flashforge_adventurer_3/protocol.py

```diff
@@ -1,30 +1,38 @@
 import asyncio
 from codecs import StreamReader, StreamWriter
 import logging
-import os
 import re
 from typing import Optional, Tuple, TypedDict
 
 logger = logging.getLogger(__name__)
 
 BUFFER_SIZE = 1024
 TIMEOUT_SECONDS = 5
-STATUS_COMMAND = '~M601 S1'
-PRINT_JOB_INFO_COMMAND = '~M27'
-TEMPERATURE_COMMAND = '~M105'
+STATUS_COMMAND = '~M119\r\n'
+PRINT_JOB_INFO_COMMAND = '~M27\r\n'
+TEMPERATURE_COMMAND = '~M105\r\n'
+
+# Regular expressions for temperature and progress
+
+TEMPERATURE_REGEX_5M_PRO = re.compile(r'T0:(\d+\.?\d*)/(\d+\.?\d*) .* B:(\d+\.?\d*)/(\d+\.?\d*)')
+PROGRESS_REGEX_5M_PRO = re.compile(r'(\d+)/(\d+)\r')
+
+TEMPERATURE_REGEX_ADVENTURER_4 = re.compile(r'CMD M105 Received.\r\nT0:(\d+)\W*/(\d+) B:(\d+)\W*/(\d+)\r\n(.*?)ok\r\n')
+PROGRESS_REGEX_ADVENTURER_4 = re.compile(r'CMD M27 Received.\r\nSD printing byte (\d+)/100\r\n.*? (\d+)/(\d+)\r\nok')
+
+TEMPERATURE_REGEX_ADVENTURER_3 = re.compile(r'CMD M105 Received.\r\nT0:(\d+)\W*/(\d+) B:(\d+)\W*/(\d+)\r\n(.*?)ok\r\n')
+PROGRESS_REGEX_ADVENTURER_3 = re.compile(r'CMD M27 Received.\r\n\w+ printing byte (\d+)/(\d+)\r\n(.*?)ok\r\n')
 
-STATUS_REPLY_REGEX = re.compile('CMD M27 Received.\r\n\w+ printing byte (\d+)/(\d+)\r\n(.*?)ok\r\n')
-TEMPERATURE_REPLY_REGEX = re.compile('CMD M105 Received.\r\nT0:(\d+)\W*/(\d+) B:(\d+)\W*/(\d+)\r\n(.*?)ok\r\n')
 
 class PrinterStatus(TypedDict):
+    model: str
```

Comment:
Let's use strong typing for `model` rather than passing in any arbitrary string. Either a string literal or an enum can work.